cdc/sink/codec/avro.go

```diff
@@ -160,30 +152,86 @@ func (a *AvroEventBatchEncoder) Size() int {
 	return sum
 }
 
-func avroEncode(table *model.TableName, manager *AvroSchemaManager, tableVersion uint64, cols []*model.Column, colInfos []rowcodec.ColInfo, tz *time.Location) (*avroEncodeResult, error) {
+func (a *AvroEventBatchEncoder) avroEncode(
+	ctx context.Context,
+	e *model.RowChangedEvent,
+	topic string,
+	key bool,
+) (*avroEncodeResult, error) {
+	var cols []*model.Column
+	var colInfos []rowcodec.ColInfo
+	var enableTiDBExtension bool
+	var schemaManager *AvroSchemaManager
+	var operation string
+	if key {
+		cols, colInfos = e.HandleKeyColInfos()
+		enableTiDBExtension = false
+		schemaManager = a.keySchemaManager
+	} else {
+		cols = e.Columns
+		colInfos = e.ColInfos
+		enableTiDBExtension = a.enableTiDBExtension
+		schemaManager = a.valueSchemaManager
+		if e.IsInsert() {
+			operation = "c"
```

Comment:
These are best if they can be constants, and if that seems too overkill, then there should at least be a comment explaining exactly how this one letter is understood.